File: py/math/prime-seive.py
import math

input_range=200
arr = []
arr = [0]*10000

s = math.sqrt(input_range)

for i in range(2,input_range+1):
    if arr[i] != 1:
        arr[i]=0
    for j in range(2, int(s+3)):
        if i*j <= input_range:
            arr[i*j]=1
        

count=0
for i in range(2,input_range+1):
    if arr[i] == 0:
        print i
        count=count+1
        
print "total: ",count

File: java/math/PrimeSeive.java
class PrimeSeive
{
	public static void main (String[] args)
	{
        int input=200;
        int[] arr = new int[10000];
    	double s = Math.sqrt(input);
    	
    	int i,j;
    	for(i=2; i<=input; i++){
    		if(arr[i] != 1){
    			arr[i]=0;
    		}
    		for(j=2; j<s+2; j++){
    			if(i*j <= input){
    				arr[i*j]=1;
    			}
    		}
    	}
    	
    	int count=0;
    	for(i=2;i<=input;i++){
    		if(arr[i] == 0){
    		    System.out.println(i);
    			count++;
    		}
    	}
    	
        System.out.println("total: "+count);

	}
}